src/Chainweb/Chainweb.hs

```diff
@@ -753,11 +753,26 @@ runChainweb cw = do
                 mw)
             (monitorConnectionsClosedByClient clientClosedConnectionsCounter)
 
-    requestSizeLimit :: Middleware
-    requestSizeLimit = requestSizeLimitMiddleware $
+    -- Request size limit for the service API
+    --
+    serviceRequestSizeLimit :: Middleware
+    serviceRequestSizeLimit = requestSizeLimitMiddleware $
         setMaxLengthForRequest (\_req -> pure $ Just $ 2 * 1024 * 1024) -- 2MB
         defaultRequestSizeLimitSettings
 
+    -- Request size limit for the P2P API
+    --
+    -- NOTE: this may need to have to be adjusted if the p2p limits for batch
+    -- sizes or number of branch bound change. It may also need adjustment for
+    -- other protocol changes, like additional HTTP request headers or changes
+    -- in the mempool protocol.
+    --
+    -- FIXME: is 8K enough for the mempool?
+    --
+    p2pRequestSizeLimit :: Middleware
+    p2pRequestSizeLimit = requestSizeLimitMiddleware $
+        setMaxLengthForRequest (\_req -> pure $ Just $ 8 * 1024) -- 8KB
```

Comment:
Would it be better to make these tunable values via the config file? Or do we want to specifically discourage that?